### ISSUE TITLE
SV1 calculates only the gradient if parameters are explicitly marked `requires_grad=True`

### ISSUE THREAD

**Issue by speller26** [aws/amazon-braket-pennylane-plugin-python#129]
**Describe the bug**
If parameters passed into a qnode are PennyLane tensors explicitly marked with `requires_grad=True`, BraketAwsQubitDevice with SV1 will only calculate measurements through the AdjointGradient result type, which in turn only supports expectation values. As another consequence, attempting to run circuits with other measurements (like var), with more than one observable, or no observables at all (like the Probability measurement) will fail altogether.

**To reproduce**
```python
In [1]: import pennylane as qml
   ...: from pennylane import numpy as qnp

In [2]: device_arn = "arn:aws:braket:::device/quantum-simulator/amazon/sv1"
   ...: dev = qml.device("braket.aws.qubit", device_arn=device_arn, wires=2, shots=0)

In [3]: weights = qnp.array([0.1, 0.2, 0.3], requires_grad=True)  # trainable

In [4]: weights2 = [0.1, 0.2, 0.3]  # not trainable

In [5]: @qml.qnode(dev)
   ...: def circuit1(a, weights):
   ...:     qml.RZ(a, wires=0)
   ...:     qml.RY(a, wires=1)
   ...:     qml.RX(weights[0], wires=0)
   ...:     qml.RY(weights[1], wires=1)
   ...:     qml.CNOT(wires=[0, 1])
   ...:     qml.RZ(3.14, wires=1)
   ...:     qml.RX(weights[0], wires=1)
   ...:     qml.RX(a, wires=1)
   ...:     return qml.expval(qml.PauliX(wires=0) @ qml.PauliY(wires=1))
   ...: 

In [6]: circuit1(3, weights)  # trainable
   ...: print(dev.circuit)
T  : |  0  |   1   |2|   3    |   4   |  5  |    Result Types    |
                                                                  
q0 : -Rz(3)-Rx(p_2)-C------------------------AdjointGradient(X@Y)-
                    |                        |                    
q1 : -Ry(3)-Ry(p_3)-X-Rz(3.14)-Rx(p_5)-Rx(3)-AdjointGradient(X@Y)-

T  : |  0  |   1   |2|   3    |   4   |  5  |    Result Types    |

Unassigned parameters: [p_2, p_3, p_5].

In [7]: circuit1(3, weights2)  # not trainable
   ...: print(dev.circuit)
T  : |  0  |   1    |2|   3    |   4    |  5  |  Result Types  |
                                                                
q0 : -Rz(3)-Rx(0.10)-C-------------------------Expectation(X@Y)-
                     |                         |                
q1 : -Ry(3)-Ry(0.20)-X-Rz(3.14)-Rx(0.10)-Rx(3)-Expectation(X@Y)-

T  : |  0  |   1    |2|   3    |   4    |  5  |  Result Types  |
```
**Expected behavior**
The device should calculate any requested measurement rather than just expectation.

**Screenshots or logs**
If applicable, add screenshots or logs to help explain your problem.

**System information**
A description of your system. Please provide:
- **Amazon Braket Python PennyLane Plugin version**: v1.10.0 (and up)
- **Amazon Braket Python SDK version**:
- **Amazon Braket Python Schemas version**:
- **Amazon Braket Python Default Simulator version**:
- **Python version**:

**Workaround**
Please use versions of the plugin <1.10.0 if you want to use tensors with `requires_grad=True`.
